src/solver/variable/state.cpp

```diff
@@ -353,117 +352,149 @@ void State::yearEndBuildFromThermalClusterIndex(const uint clusterAreaWideIndex,
     }
 
     if (dur > 0)
+        yearEndBuildFromThermalClusterIndexDurPositive(dur, ON_min, ON_max, ON_opt);
```

Comment:
function **yearEndBuildFromThermalClusterIndexDurPositive** : 
I don't understand the details of what it does but what I see is that its mission is to populate **ON_opt**, and that the reason why tests were failing when passing it without reference.
As mentioned when declaring  **ON_opt**, it's content is : 
_Nombre de groupes économiquement optimal en fonctionnement à l'heure h_
So we should : 
- rename this function, for instance, into : **computeEconomicallyOptimalNbClustersONforEachHour**
- make this function return the filled **std::array** : 
```C++
if (dur > 0)
    ON_opt = computeEconomicallyOptimalNbClusterForEachHour(dur, ON_min, ON_max);
```
I think this should not be costly at execution, as the move semantics will be triggered automatically, right ?.

Reply:
By the way , according to comments coming with its declaration, local variable **dur** should be renamed into **maxDurationON** (for instance).